datm/datm_datamode_simple_mod.F90

```diff
@@ -0,0 +1,385 @@
+  real(r8), pointer :: Faxa_lwdn(:)  => null()
+  real(r8), pointer :: Faxa_rainc(:) => null()
+  real(r8), pointer :: Faxa_rainl(:) => null()
+  real(r8), pointer :: Faxa_snowc(:) => null()
+  real(r8), pointer :: Faxa_snowl(:) => null()
+  real(r8), pointer :: Faxa_swndr(:) => null()
+  real(r8), pointer :: Faxa_swndf(:) => null()
+  real(r8), pointer :: Faxa_swvdr(:) => null()
+  real(r8), pointer :: Faxa_swvdf(:) => null()
+  real(r8), pointer :: Faxa_swnet(:) => null()
+  real(r8), pointer :: Faxa_ndep(:,:) => null()
+
+  ! stream data
+  real(r8), pointer :: strm_swdn(:)      => null()
+
+  ! othe module arrays
+  real(R8), pointer :: yc(:)                 ! array of model latitudes
+
+  ! constant forcing values
+  real(R8) :: dn10 = 1.204_R8
+  real(R8) :: slp = 101325.0_R8
+  real(R8) :: q = 0.0_R8
+  real(R8) :: t = 273.15_R8
+  real(R8) :: u = 0.0_R8
+  real(R8) :: v = 0.0_R8
+
+  ! constants
+  real(R8) , parameter :: tKFrz    = SHR_CONST_TKFRZ
+  real(R8) , parameter :: degtorad = SHR_CONST_PI/180.0_R8
+  real(R8) , parameter :: phs_c0   =   0.298_R8
+  real(R8) , parameter :: dLWarc   =  -5.000_R8
+
+  character(*), parameter :: nullstr = 'null'
+  character(*), parameter :: rpfile  = 'rpointer.atm'
+  character(*), parameter :: u_FILE_u = &
+       __FILE__
+
+!===============================================================================
+contains
+!===============================================================================
+
+  subroutine datm_datamode_simple_advertise(exportState, fldsexport, flds_scalar_name, &
+    nlfilename, my_task, mpicom, rc)
+
+    ! input/output variables
+    type(esmf_State)   , intent(inout) :: exportState
+    type(fldlist_type) , pointer       :: fldsexport
+    character(len=*)   , intent(in)    :: flds_scalar_name
+    character(len=*)   , intent(in)    :: nlfilename
+    integer            , intent(in)    :: my_task
+    integer            , intent(in)    :: mpicom
+    integer            , intent(out)   :: rc
+
+    ! local variables
+    type(fldlist_type), pointer   :: fldList
+    integer           , parameter :: main_task   = 0 ! task number of main task
+    integer                       :: ierr       ! error code
+    integer                       :: nu         ! unit number
+    character(len=*)  , parameter :: subname='(datm_datamode_simple_advertise): '
+
+    !-------------------------------------------------------------------------------
+
+    namelist / const_forcing_nml / dn10, slp, q, t, u, v
+
+    rc = ESMF_SUCCESS
+
+    ! Read const_forcing_nml from nlfilename
+    if (my_task == main_task) then
+       open (newunit=nu,file=trim(nlfilename),status="old",action="read")
+       read (nu,nml=const_forcing_nml,iostat=ierr)
+       close(nu)
+       if (ierr > 0) then
+          call shr_sys_abort(subName//': namelist read error '//trim(nlfilename))
+       end if
+    end if
+
+    call shr_mpi_bcast(dn10         , mpicom, 'dn10')
```

Comment:
Can we move away from shr_mpi_bcast and use ESMF_VMBroadcast instead?

Reply:
https://earthsystemmodeling.org/docs/release/latest/ESMF_refdoc/node6.html#SECTION060126000000000000000

Reply:
@alperaltuntas Note that as of #203 CDEPS no longer uses shr_mpi.